src/libraries/System.Private.CoreLib/src/System/Threading/EventWaitHandle.cs

```diff
@@ -10,28 +10,102 @@ namespace System.Threading
 {
     public partial class EventWaitHandle : WaitHandle
     {
-        public EventWaitHandle(bool initialState, EventResetMode mode) :
-            this(initialState, mode, null, out _)
+        public EventWaitHandle(bool initialState, EventResetMode mode)
         {
+            CreateEventCore(initialState, mode, name: null, options: default, out _);
```

Comment:
For trim friendliness, can we just do the simple create event here to avoid dragging all the extra named machinery into typical trimmed or native AOT apps? (Same for Semaphore and Mutex.)

Reply:
Yea it would be good to be able to trim out the named machinery when names aren't used, I'll look into refactoring the common code to help with trimming.

Reply:
> I'll look into refactoring the common code to help with trimming.

Nit: A small amount of duplication to keep things simple is fine me.